### PR TITLE
Add a tx hash when getting an unsigned tx

### PR DESCRIPTION
**Pull request by velzevur**
[Pivotal story](https://www.pivotaltracker.com/story/show/155320694)
When a user gets an unsigned tx, one also receives a tx hash: the hash of the tx if it is signed and posted to the chain so its state could be tracked.
CT logs:
```
POST "http://127.0.0.1:3013/v2/tx/spend", type "application/json", Body <<"
{\"amount\":2,
\"fee\":1,
\"nonce\":661,
\"recipient_pubkey\":\"ak$2cYML1pEoMQBy4KE3fYtxUBuAwx3aG891TEKcg8VHn2721CdAe\",
\"sender\":\"ak$3U1i4g28WiNPy2H5fqm1JtSogC7uP1dA5KyX8Q1rLVhWfyZezUMAmdU7DC8Aa11G7TDvqLLDebvxw2JUj45nLk3NNzKj42\"}">>

Return code 200, Body "
{\"tx\":\"tx$3WbJe3wvTXn2QJjHyuywjpXLLcduzfq7QfqB8khh268B8E249s4V8mjevVX8KW8TwKTutbLr8vuAXdtg3eLUkRb1And4NpZ6xQKyiiRQ9cQ32p5fCG1X3Pa3ZVzJMjhfzbahyRWb12YMgQPMP7UA98gNXwTJGefFdG58ABWugSCUCBFKbsUqtEXCe9Co4mqQs5D7Ck6voR5m4YyeemcxqRmUf6woh9iaz2RiMdZvYwMfSbMDb\",
\"tx_hash\":\"th$2LbYF5jXBDELz2uUzNYwugeNNQYuxYEc3ubNubeiDEyxrx6oyf\"}"
```

Also some HTTP helpers optimization

### CODE REVIEW

**Comment by lucafavatella** [comment]
@velzevur Test failed
```
%%% aecore_forking_SUITE ==> {{badmatch,
     {{block,1,
```

It looks like the perfect opportunity for reviewing PR #774 that aims at fixing it.

File: apps/aehttp/src/aehttp_helpers.erl [hunk 1]
```diff
@@ -187,7 +194,7 @@ nameservice_pointers_decode(PointersKey) ->
         Pointers = maps:get(PointersKey, State),
         try {ok, maps:put(PointersKey, jsx:decode(Pointers), State)}
         catch _:_ ->
-            {error, {400, [], #{<<"reason">> => <<"Invalid pointers">>}}}
+            {error, {400, [], #{<<"reason">> => <<"Invayylid pointers">>}}}
```
Comment (lucafavatella) [inline apps/aehttp/src/aehttp_helpers.erl]: Typo in reason returned to user.

Comment (velzevur) [reply apps/aehttp/src/aehttp_helpers.erl]: auch

**Comment by velzevur** [comment]
Fixed typo and added proper tests for nameservice pointers (that actually tests the message with a typo)

**Comment by velzevur** [comment]
Rebased on top of #770 changes.

File: apps/aehttp/src/aehttp_helpers.erl [hunk 2]
```diff
@@ -14,23 +14,30 @@
...
+-export([ ok_response/1
+        , unsigned_tx_response/1]).
 
-parse_request([], _Req, Result) ->
-    {ok, Result};
-parse_request([Fun | T], Req, Result0) ->
+process_request(FunsList, Req) ->
+    process_request(FunsList, Req, #{}).
+
+process_request([], _Req, {ok, {Code, _, _Response} = Result})
```
Comment (uwiger) [inline apps/aehttp/src/aehttp_helpers.erl]: It seems to me as if this clause is redundant, since line 29 will also match and produce the same result.

**Pull request merged**